### ISSUE TITLE
bug(YouTube/Spoof streaming data): playback issues occur when data connection changes or RVX has been open for a long time

### ISSUE THREAD

**Issue by ArcherLegend** [inotia00/ReVanced_Extended#2416]
### Type

Error at runtime

### Tools used

rvx-builder

### Application

Youtube 19.16.39
Patches versions have had this issue for the last few months. The issue is still present with release 4.14.1 and 4.14.2-dev1

### Bug description

After RVX is started for the first time when turning on the phone (or after not being used for a few hours) if you start playing a video it buffers without loading. The workarounds are the following 2:

close RVX and open it again
disconnect from wifi/data and reconnect
Once that is done every video starts playing normally till the device is rebooted or RVX is not used for a few hours.

Spoof player parameters is turned on and the chosen client (iOS, Android TV, Android VR) does not fix the issue (my settings are attached in the additional context section).
The bug appears whether MicroGMS is restricted or unrestricted.

After cleanly installing both MicroGMS and RVX the issue may not appear instantly but it may take 1 or 2 days before occuring and once that happens for the first time it keeps doing what I described.

### Additional context

"append_time_stamp_information": false,
"default_video_quality_mobile": 1080,
"default_video_quality_wifi": 1080,
"disable_ambient_mode": true,
"disable_ambient_mode_in_fullscreen": true,
"disable_hdr_video": true,
"disable_speed_overlay": true,
"disable_splash_animation": true,
"enable_gradient_loading_screen": true,
"enable_seekbar_tapping": false,
"enable_swipe_brightness": false,
"enable_swipe_lowest_value_auto_brightness": false,
"enable_swipe_press_to_engage": true,
"enable_watch_panel_gestures": true,
"hide_carousel_shelf": true,
"hide_clip_button": true,
"hide_comment_create_shorts_button": true,
"hide_comment_thanks_button": true,
"hide_comment_timestamp_and_emoji_buttons": true,
"hide_comments_by_members": true,
"hide_download_button": true,
"hide_feed_survey": false,
"hide_handle": false,
"hide_latest_posts": false,
"hide_latest_videos_button": false,
"hide_mix_playlists": true,
"hide_movie_shelf": true,
"hide_navigation_notifications_button": true,
"hide_navigation_shorts_button": true,
"hide_notify_me_button": true,
"hide_player_autoplay_button": false,
"hide_player_youtube_music_button": true,
"hide_quick_actions_more_button": true,
"hide_quick_actions_open_mix_playlist_button": true,
"hide_quick_actions_open_playlist_button": true,
"hide_remix_button": true,
"hide_rewards_button": true,
"hide_shop_button": true,
"hide_shorts_shelf_history": true,
"hide_snack_bar": true,
"hide_thanks_button": true,
"hide_toolbar_create_button": true,
"hide_voice_search_button": true,
"remember_playback_speed_last_selected": false,
"remove_viewer_discretion_dialog": true,
"swipe_overlay_background_alpha": 0,
"swipe_overlay_rect_size": 50,
"switch_create_with_notifications_button": false,
"ryd_user_id": "act0lY7slv00JLqJpNEYa8TvpBBAhoN6fIKr"

### Device Environment

Galaxy S10+ (Exynos 9820 variant) Android 10 One UI 2.1 build version QP1A.190711.020.G975FXXU8CTG4

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I didn't use any settings marked as `Experimental Flags`.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by 1node0001** [comment]
Your phone is super old also, almost museum grade old. I havent had a single buffer since I installed RVX 4 months ago, but I dont login at all, maybe thats why.

**Comment by inotia00** [comment]
On my side, the issue is not reproduced

Check the following and report back the results

1. Does the issue reproduce on unpatched YouTube 19.16.39?

2. Does the issue reproduce if you include only the `GmsCore support` patch and exclude all other patches?

3. Does the issue reproduce if you turn off the `Spoof streaming data` setting?

**Comment by inotia00** [comment]
> "gms_show_dialog": false,

These settings are not in my source

Are you sure you are using my source?

**Comment by ArcherLegend** [comment]
> > "gms_show_dialog": false,
> 
> These settings are not in my source
> 
> Are you sure you are using my source?

Sorry I mistankely copied my RVX settings from anddea instead of the ones which I am using on your RVX fork, I made the same thread on anddea's Github page but I was told to refer to your's as he is using your codebase.

**Comment by ArcherLegend** [comment]
> On my side, the issue is not reproduced
> 
> Re-patch to the dev branch of my patch
> 
> 1. Check if the issue persists when you turn off `Spoof streaming data`
> 2. Check if the issue persists when you switch to the `Android VR` client

Later today I will repatch with your dev branch and try what you said however I can already tell you that with the last few releases I have already tried point 1 and 2 with your RVX fork but the issue still happens.
I will try the dev patch ASAP and reach back to you.

**Comment by ArcherLegend** [comment]
> > "gms_show_dialog": false,
> 
> These settings are not in my source
> 
> Are you sure you are using my source?

I have mistakenly copied and pasted anddea's 2.229.0 settings as I am also using his version of RVX alongside yours. I have now edited the report with the settings I use for your Revanced fork.
I also opened this same thread on anddea's GitHub page but he told me to refer to you as he is using your codebase.

**Comment by ArcherLegend** [comment]
> On my side, the issue is not reproduced
> 
> Re-patch to the dev branch of my patch
> 
> 1. Check if the issue persists when you turn off `Spoof streaming data`
> 2. Check if the issue persists when you switch to the `Android VR` client

I have cleanly repatched your latest dev build and if I turn off the "Spoof streaming data" the videos won't ever start.
If I switch to iOS spoofing I get the issue I described in this bug report.
In the past I have already tried Android VR but I had this issue. I am spoofing to Android VR now and for the moment the issue is not happening anymore, however note that usually it can start to happen 1 or 2 days after the spoofing is changed / RVX is cleanly installed so before marking this bug report as closed wait for me to reach back in the next day or 2 as I think that the bug may present again despite spoofing to Android VR.

**Comment by ArcherLegend** [comment]
As a non related note I have noticed that despite me not including Alternative Thumbnails patch when using rvx-builder the patch is applied (I have already cleared every temporary folder / setting file related to rvx-builder before patching many times and each time I get this behaviour).
In am on Windows 10.

**Comment by inotia00** [comment]
Thanks for your response

The reason I asked for this test is because I didn't know which patch was causing the issue

Also, in the latest dev branch of my patch, there was one test code applied related to the `Spoof streaming data` patch

This is the code that solves the exact same issue you reported, and it is only applied to Android VR client and Android TV client so far

Please test it with Android VR client or Android TV client for about a day and report back the results

If it is confirmed that there is no problem, I will apply this to iOS client as well

**Comment by ArcherLegend** [comment]
Ok, I'll report back tomorrow (more or less 30 hours from now)

**Comment by ArcherLegend** [comment]
@inotia00 for now I can confirm you that the issue is not reproducing anymore with the latest branch and the spoofing to Android VR. I will reach back in this bug report if the situation changes but for now I would consider this bug solved in your latest dev branch.

May I ask you if you plan on adding a "don't show this again" option for the GmsCore optimisation reminder when you open RVX like the one of Anddea's RVX branch?

Thank you

**Comment by YT-Advanced** [comment]
> May I ask you if you plan on adding a "don't show this again" option for the GmsCore optimisation reminder when you open RVX like the one of Anddea's RVX branch?

Already implemented

**Comment by ArcherLegend** [comment]
> > May I ask you if you plan on adding a "don't show this again" option for the GmsCore optimisation reminder when you open RVX like the one of Anddea's RVX branch?
> 
> Already implemented

May I ask you were I can find it? Because on anddea's version you have the option both on the popup and in the RVX settings, whereas on inotia00's version there is no such option in the popup and I cannot find it in the RVX settings.

**Comment by YT-Advanced** [comment]
> May I ask you were I can find it? Because on anddea's version you have the option both on the popup and in the RVX settings, whereas on inotia00's version there is no such option in the popup and I cannot find it in the RVX settings.

Uhm in gmscore support settings, turn off check gmscore
![Screenshot_20241003-155041_RVX Manager.jpg](https://github.com/user-attachments/assets/ac641f36-a561-4a11-822b-c849a66fb695)

**Comment by ArcherLegend** [comment]
Oh I see so I need to disable that when patching, I usually patch with RVX-builder so I will have to modify the options.json I guess.
Perhaps having an in app setting like with Anddea's versions may be better if it is not too difficoult to implent.
Thank you

**Comment by inotia00** [comment]
fixed in [revanced-patches-4.14.2](https://github.com/inotia00/revanced-patches/releases/tag/v4.14.2)
